### PR TITLE
feat: add a new provider fireworks ai

### PR DESCRIPTION
**Pull request by LinkW77**
# Pull Request

## PR Description

Add a new provider fireworks ai

## Linked Issue

Integrate Fireworks AI Models with TaskingAI #297

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Performance enhancement
- [ ] Code refactor
- [ ] Documentation update
- [ ] Other, please describe:

## Checklist

Before submitting this PR, please make sure:

- [x] I have read the [CONTRIBUTING.md](../CONTRIBUTING.md) guidelines.
- [x] I have tested my changes locally to ensure they are effective.
- [x] I have updated the necessary documentation (if applicable).

### CODE REVIEW

**Review by SimsonW — DISMISSED** [review]
LGTM

File: inference/providers/fireworks/text_embedding.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        input_type: Optional[TextEmbeddingInputType] = None,
+        proxy: Optional[str] = None,
+        custom_headers: Optional[Dict[str, str]] = None,
+    ) -> TextEmbeddingResult:
+        api_url = "https://api.fireworks.ai/inference/v1/embeddings"
+
+        headers = _build_fireworks_header(credentials)
+
+        payload = {"model": provider_model_id, "input": input}
+        if proxy:
```
Comment (jameszyao) [inline inference/providers/fireworks/text_embedding.py:21]: proxy not supported

**Review by SimsonW — APPROVED** [review]
LGTM

**Review by taskingaijc — APPROVED** [review]
LGTM

**Pull request merged**